base_name_search_improved/models/ir_model.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# © 2016 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp import SUPERUSER_ID
+from openerp import tools
+
+
+# Extended name search is only used on some operators
+ALLOWED_OPS = set(['ilike', 'like'])
+
+
+@tools.ormcache(skiparg=0)
```

Comment:
Ah sorry, you're right. I started using it in v9 and got used to the new system.